tron/core/jobrun.py

```diff
@@ -210,43 +211,45 @@ def _start_action_runs(self):
         """Start all startable action runs, and return any that were
         successfully started.
         """
-        started_actions = []
-        for action_run in self.action_runs.get_startable_action_runs():
-            if action_run.start():
-                started_actions.append(action_run)
+        return [
+            action_run
+            for action_run in self.action_runs.get_startable_action_runs()
+            if action_run.start()
+        ]
 
-        return started_actions
-
-    def handle_action_run_state_change(self, action_run, _):
+    def handle_action_run_state_change(self, action_run: ActionRun, _):
         """Handle events triggered by JobRuns."""
         # propagate all state changes (from action runs) up to state serializer
         self.notify(self.NOTIFY_STATE_CHANGED)
 
         if not action_run.is_done:
             return
 
+        if not action_run.is_broken and action_run.trigger_downstreams:
```

Comment:
We can put the logic about not action_run.is_broken together, so it would be easier to read.